### PR TITLE
[Accordion] Library styling provides default outline for accessibility

### LINKED ISSUE #1782: [Accessibility][Accordion] Missing visual keyboard focus on the accordion headings/trigger buttons
**Steps to Reproduce**
1. Open page in Google Chrome.
2. Use TAB and/or TAB + SHIFT to navigate through the accordion component headings.

**Current Behavior**
When focused, the accordion headings provide no discernible visual indication of focus, because the CSS `outline` property's value is set to `none`. It is difficult for sighted keyboard users to understand where their keyboard focus is and how to interact with their intended control.

**Expected behavior/code**
Actionable/focusable items should provide a visual indicator of focus. When the default focus indicator is removed (e.g. with `outline: none`), it must be replaced with a focus indicator that provides at least a 3:1 contrast ratio with the background colors.

**Environment**
Chrome 92, Windows 10
AEM Core Components v. 2.17  

**Affected Users**
Sighted keyboard users

**WCAG SC**
2.4.7 Focus Visible

### PR DESCRIPTION
**Pull request by bartoszglow**
fixes #1782

Fixes demo component overflowing on mobile in components library.

### CODE REVIEW

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less [hunk 1]
```diff
@@ -35,4 +35,5 @@
 
 .cmp-examples-demo__component--widthFull {
     width: 100%;
+    max-width: 100%;
```
Comment (comanV) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less:38]: Isn't redundant to have `width: 100%` and `max-width: 100%` on the same element?

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-site/styles/components/demo.less [hunk 2]
```diff
@@ -46,6 +46,7 @@
 
 .cmp-examples-demo__component--widthFull {
     width: 100%;
+    max-width: 100%;
```
Comment (comanV) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-site/styles/components/demo.less:49]: Isn't redundant to have `width: 100%` and `max-width: 100%` on the same element?

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less [hunk 3]
```diff
@@ -40,7 +40,6 @@
         color: inherit;
         font: inherit;
         cursor: pointer;
-        outline: inherit;
```
Comment (comanV) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less:43]: Shoud be `outline: inherit;` deleted from here too?
https://github.com/adobe/aem-core-wcm-components/blob/c6756447df4ebcfd66d133385502cae3135f87be/examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less#L43

Comment (comanV) [inline examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less:43]: Should `outline: inherited` be deleted from here too?
https://github.com/adobe/aem-core-wcm-components/blob/c6756447df4ebcfd66d133385502cae3135f87be/examples/ui.apps/src/content/jcr_root/apps/core-components-examples/components/accordion/clientlibs/amp/css/accordion.less#L57

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less (hunk 1 above)
Comment (bartoszglow) [reply examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less:38]: It is not. This case covers both cases:
- if content of demo component is wider than width of its container then it gets narrowed by `max-width: 100%`
- if content of demo component is narrower than width of its container then it gets expanded by `width: 100%`

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less (hunk 3 above)
Comment (bartoszglow) [reply examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-themes/core-components-clean/styles/components/accordion/base.less:43]: AMP mode is technically available for screen readers. That is probably very small percentage of use case however we should cover the same accessibility fix for both AMP and standard pages.

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less (hunk 1 above)
Comment (comanV) [reply examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less:38]: > `if content of demo component is wider than width of its container then it gets narrowed by max-width: 100%`

If a child element is wider than the width of its container it gets narrowed by `width:100%`. I put a small screenshot of a basic example.
Screenshot 2022-01-14 at 17 51 58

**Review by comanV — APPROVED** [review]

File: examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less (hunk 1 above)
Comment (bartoszglow) [reply examples/ui.apps/src/content/jcr_root/apps/core-components-examples/clientlibs/clientlib-base-amp/styles/base/demo.less:41]: Addressed. `max-width` provided only for `--widthAuto` variant of demo component so it prevents some iframes from getting out of 100% width of component.

**Pull request merged**